From fcc756140c9575d591e1c314012b1c595770f2f7 Mon Sep 17 00:00:00 2001
From: Puunithaaraj Gopal <puunithaaraj.gopal@intel.com>
Date: Fri, 24 May 2019 20:23:37 +0800
Subject: [PATCH 1136/1225] media: ici: Use ICI_FORMAT_RGB24 for packed formats

ICI_FORMAT_RGB24 format as packed format handling

Tracked-On: HSD-1507216987
Tracked-On: PKT-2253
Signed-off-by: Puunithaaraj Gopal <puunithaaraj.gopal@intel.com>
---
 drivers/media/pci/intel/ici/ici-isys-csi2.c   | 2 ++
 drivers/media/pci/intel/ici/ici-isys-stream.c | 1 -
 drivers/media/pci/intel/ici/ici-isys-subdev.c | 2 ++
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ici/ici-isys-csi2.c b/drivers/media/pci/intel/ici/ici-isys-csi2.c
index d27119a..f216a93 100644
--- a/drivers/media/pci/intel/ici/ici-isys-csi2.c
+++ b/drivers/media/pci/intel/ici/ici-isys-csi2.c
@@ -19,6 +19,7 @@
 
 static const uint32_t ici_csi2_supported_codes_pad_sink[] = {
 	ICI_FORMAT_RGB888,
+	ICI_FORMAT_RGB24,
 	ICI_FORMAT_RGB565,
 	ICI_FORMAT_UYVY,
 	ICI_FORMAT_YUYV,
@@ -43,6 +44,7 @@ static const uint32_t ici_csi2_supported_codes_pad_sink[] = {
 
 static const uint32_t ici_csi2_supported_codes_pad_source[] = {
 	ICI_FORMAT_RGB888,
+	ICI_FORMAT_RGB24,
 	ICI_FORMAT_RGB565,
 	ICI_FORMAT_UYVY,
 	ICI_FORMAT_YUYV,
diff --git a/drivers/media/pci/intel/ici/ici-isys-stream.c b/drivers/media/pci/intel/ici/ici-isys-stream.c
index d8c403e..b2090b1 100644
--- a/drivers/media/pci/intel/ici/ici-isys-stream.c
+++ b/drivers/media/pci/intel/ici/ici-isys-stream.c
@@ -74,7 +74,6 @@ const struct ici_isys_pixelformat ici_isys_pfmts_be_soc[] = {
 };
 
 const struct ici_isys_pixelformat ici_isys_pfmts_packed[] = {
-
 	{ ICI_FORMAT_UYVY, 16, 16, ICI_FORMAT_UYVY, IA_CSS_ISYS_FRAME_FORMAT_UYVY },
 	{ ICI_FORMAT_YUYV, 16, 16, ICI_FORMAT_YUYV, IA_CSS_ISYS_FRAME_FORMAT_YUYV },
 	{ ICI_FORMAT_RGB565, 16, 16, ICI_FORMAT_RGB565, IA_CSS_ISYS_FRAME_FORMAT_RGB565 },
diff --git a/drivers/media/pci/intel/ici/ici-isys-subdev.c b/drivers/media/pci/intel/ici/ici-isys-subdev.c
index 4d12a70..b727ba3 100644
--- a/drivers/media/pci/intel/ici/ici-isys-subdev.c
+++ b/drivers/media/pci/intel/ici/ici-isys-subdev.c
@@ -12,6 +12,7 @@
 unsigned int ici_isys_format_code_to_bpp(u32 code)
 {
 	switch (code) {
+	case ICI_FORMAT_RGB24:
 	case ICI_FORMAT_RGB888:
 		return 24;
 	case ICI_FORMAT_RGB565:
@@ -47,6 +48,7 @@ unsigned int ici_isys_format_code_to_mipi(u32 code)
 	switch (code) {
 	case ICI_FORMAT_RGB565:
 		return ICI_ISYS_MIPI_CSI2_TYPE_RGB565;
+	case ICI_FORMAT_RGB24:
 	case ICI_FORMAT_RGB888:
 		return ICI_ISYS_MIPI_CSI2_TYPE_RGB888;
 	case ICI_FORMAT_UYVY:
-- 
2.7.4

